### PR TITLE
CODENVY-1271: refactor projects backuping

### LINKED ISSUE #1271: All bitnami stacks do not work on the nightly server
**Reproduction Steps:**
* Go to  the https://nightly.codenvy-stg.com
* Try to create a template project from any  bitnami stack in the stack library.

**Expected behavior:**
* The project should be created and opened in the IDE.

**Observed behavior:**
* The workspace agent does not start.

**Codenvy version:** 5.0.0-M8
**OS and version:** Ubuntu 14.04

**Additional information:**
* Problem can be reliably reproduced: Yes
* See the attachment:
bitnami_stack

### PR DESCRIPTION
**Pull request by garagatyi**
### What does this PR do?
Refactor projects backuping.
Detect user in dev-machine container to backup projects.

### What issues does this PR fix or reference?
Fixes #1271 

### Previous Behavior
Restoring of workspaces created from images without rsync failed.

### New Behavior
Such workspaces starts well.

### Tests written?
Yes

### CODE REVIEW

File: plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/backup/WorkspaceFsBackupScheduler.java [hunk 1]
```diff
@@ -43,112 +41,98 @@
...
+        this.lastWorkspaceSynchronizationTime = new ConcurrentHashMap<>();
+        this.workspacesBackupsInProgress = new ConcurrentHashMap<>();
     }
 
     @ScheduleRate(initialDelay = 1, period = 1, unit = TimeUnit.MINUTES)
     public void scheduleBackup() {
         for (Map.Entry<String, WorkspaceRuntimes.WorkspaceState> wsStateEntry :
                 workspaceRuntimes.getWorkspaces().entrySet()) {
 
+            String workspaceId = wsStateEntry.getKey();
```
Comment (sleshchenko) [inline plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/backup/WorkspaceFsBackupScheduler.java]: I don't see usage of `wsStateEntry.getValue()`. Would it better to use `workspaceRuntimes.getWorkspaces().keySet()` for loop?

File: plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/backup/DockerEnvironmentBackupManager.java [hunk 2]
```diff
@@ -0,0 +1,550 @@
...
+            String srcUserName = getUserName(workspaceId, containerId).name;
+
+            backupAndCleanupInsideLock(workspaceId,
+                                       srcPath,
+                                       nodeHost,
+                                       syncPort,
+                                       srcUserName,
+                                       destPath);
+        } catch (IOException | ServerException e) {
+            throw new ServerException(e.getLocalizedMessage(), e);
```
Comment (sleshchenko) [inline plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/backup/DockerEnvironmentBackupManager.java]: Why you should catch ServerException to rethrow it as ServerException?

Comment (garagatyi) [inline plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/backup/DockerEnvironmentBackupManager.java]: I'll fix, thanks

File: plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/backup/WorkspaceFsBackupScheduler.java (hunk 1 above)
Comment (garagatyi) [inline plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/backup/WorkspaceFsBackupScheduler.java]: ok, Ill fix that. Thanks.

**Pull request merged**